# Exercise 1
It can be calculated using $\tan(\theta) = \frac{\text{opposite}}{\text{adjacent}}=\frac{b}{a}$.

Thus: $$\theta=\arctan\left(\frac{b}{a}\right)$$

In [3]:
import math

a = 10
b = 3
math.atan2(a, b) * 180 / math.pi

73.30075576600638

Thus $73.3\degree$

# Exercise 2

In [4]:
def camera_b_distance(f, g):
    """
    camera_b_distance returns the distance (b) where the CCD should be placed
    when the object distance (g) and the focal length (f) are given
    :param f: Focal length
    :param g: Object distance
    :return: b, the distance where the CCD should be placed
    """
    # 1/g + 1/b = 1/f
    # 1/b = 1/f - 1/g
    # 1 = (1/f - 1/g)*b
    # b = 1 / (1/f - 1/g)
    return 1 / (1/f - 1/g)

We will be using mm

In [5]:
f = 15 # mm
g = [100, 1000, 5000, 15000] # mm
b = {g_: camera_b_distance(f, g_) for g_ in g}
b

{100: 17.647058823529413,
 1000: 15.228426395939087,
 5000: 15.045135406218657,
 15000: 15.015015015015017}

**It should be placed at respectively 17.65, 15.23, 15.05 and 15.02 mm.**

When object distance is increased the position of CCD gets decreasingly smaller.

# Exercise 3

1. 
$$\frac{1}{g} + \frac{1}{b} = \frac{1}{f}$$
$$f=5\mathrm{mm},\quad g=5000\mathrm{mm}$$
$$\Rightarrow b=\frac{1}{\frac{1}{f}-\frac{1}{g}} = 5.005\mathrm{mm}$$

2. 
$$\frac{b}{B}=\frac{g}{G}$$
$$\Rightarrow B=\frac{b}{\frac{g}{G}} = \frac{Gb}{g} = 1.8018\mathrm{mm}$$

3. 
$$\frac{6.4\mathrm{mm}}{640\mathrm{pixel}} =0.01\mathrm{mm}/\mathrm{pixel}$$

4. 
$$1.8018\mathrm{mm}/(0.01\mathrm{mm}/\mathrm{pixel}) = 180.18\mathrm{pixel}$$

5. 
$$\mathrm{HFOV}=2\arctan\left( \frac{\text{sensor width}}{2f} \right) = 2\arctan\left( \frac{6.4}{10} \right)$$
$$\mathrm{VFOV}=2\arctan\left( \frac{\text{sensor height}}{2f} \right) = 2\arctan\left( \frac{4.8}{10} \right)$$

In [6]:
print("HFOV = " + str(2*math.atan(6.4/10)*180/math.pi) + " degrees")
print("VFOV = " + str(2*math.atan(4.8/10)*180/math.pi) + " degrees")

HFOV = 65.23848614238565 degrees
VFOV = 51.28201164861056 degrees
